javascript/newer-JS-features.md

```diff
@@ -0,0 +1,219 @@
+# Modern Features of JavaScript
+
+### Projected Time
+
+About 2 hours
+
+- 45 minutes for the video walkthrough of the slides
+- 20 minutes for the Group Practice
+- 10 minutes for Check for Understanding
+
+### Prerequisites
+
+- Techtonica's JS workshop
+- [Basic JS practice](https://github.com/Techtonica/curriculum/blob/main/javascript/basic-js-practice.md)
+
+
+### Motivation
+
+ECMAScript 2015, or ES6 introduced many important and useful features that modernized the JavaScript language and helps us to speed the development process. 
+
+### Objectives
+
+**Participants will be able to:**
+
+- Take advantage of ES6's new defalut params.
+- Use spread operator with arrays, functions and objects.
+- Know the difference between spread and rest operator. 
+- Use of Destructured Assignment
```

Comment:
```suggestion
- Use of Destructuring Assignment
```
I've only heard it phrased as "Destructuring Assignment": https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment